overlord/exportstate/manifest.go

```diff
@@ -0,0 +1,203 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package exportstate
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+// Manifest describes content content exported to snaps or the host.
+type Manifest struct {
+	ExportedName    string `json:"exported-name"`
+	ExportedVersion string `json:"exported-version"`
+
+	SnapInstanceName string `json:"snap-instance-name"`
```

Comment:
given that everything but core and host will export to this,  I don't think we want ExportedVersion and ExportedName anymore.  Instead I think we should have a special field only for the special case:

ExportedForSnapdAsVersion ... omitempty

if this is set it means export name is "snapd" and this is the version to use.

Reply:
Hmmm, I think this will work. Let me give it a try.

Reply:
This is done now.